test/functional/testlib.bash

```diff
@@ -0,0 +1,1015 @@
+		# if the local_bundle flag is set, copy the files to the target-dir as if the
+		# bundle had been locally installed
+		if [ "$local_bundle" = true ]; then
+			sudo mkdir -p "$target_path$(dirname "$val")"
+			sudo cp "$bundle_file" "$target_path$val"
+		fi 
+	done
+	
+	# 4) Create the requested link(s) in the bundle
+	for val in "${link_list[@]}"; do
+		# if the directory the link is doesn't exist,
+		# add it to the bundle (except if the directory is "/")
+		fdir=$(dirname "$val")
+		if [ "$fdir" != "/" ]; then
+			if [ ! "$(sudo cat "$manifest" | grep -x "D\\.\\.\\..*$fdir")" ]; then
+				bundle_dir=$(create_dir "$files_path")
+				add_to_manifest "$manifest" "$bundle_dir" "$fdir"
+			fi
+		fi
+		bundle_link=$(create_link "$files_path")
+		sudo tar -rf "$version_path"/pack-"$bundle_name"-from-0.tar "$bundle_link"
+		add_to_manifest "$manifest" "$bundle_link" "$val"
+		# Add the file pointed by the link to the zero pack of the bundle
+		pfile=$(basename "$(readlink -f "$bundle_link")")
+		sudo tar -rf "$version_path"/pack-"$bundle_name"-from-0.tar "$files_path"/"$pfile"
+		if [ "$DEBUG" == true ]; then
+			echo "link -> $bundle_link"
+			echo "file pointed to -> $(readlink -f "$bundle_link")"
+		fi
+		if [ "$local_bundle" = true ]; then
+			sudo mkdir -p "$target_path$(dirname "$val")"
+			# if local_bundle is enabled copy the link to target-dir but also
+			# copy the file it points to
+			pfile_path=$(awk "/$(basename $pfile)/"'{ print $4 }' "$manifest")
+			sudo cp "$files_path"/"$pfile" "$target_path$pfile_path"
+			sudo ln -rs "$target_path$pfile_path" "$target_path$val"
+		fi
+	done
+	
+	# 5) Create the requested dangling link(s) in the bundle
+	for val in "${dangling_link_list[@]}"; do
+		# if the directory the link is doesn't exist,
+		# add it to the bundle (except if the directory is "/")
+		fdir=$(dirname "$val")
+		if [ "$fdir" != "/" ]; then
+			if [ ! "$(sudo cat "$manifest" | grep -x "D\\.\\.\\..*$fdir")" ]; then
+				bundle_dir=$(create_dir "$files_path")
+				add_to_manifest "$manifest" "$bundle_dir" "$fdir"
+			fi
+		fi
+		# Create a link passing a file that does not exits
+		bundle_link=$(create_link "$files_path" "$files_path"/"$(generate_random_name does_not_exist-)")
+		sudo tar -rf "$version_path"/pack-"$bundle_name"-from-0.tar "$bundle_link"
+		add_to_manifest --skip-validation "$manifest" "$bundle_link" "$val"
+		# Add the file pointed by the link to the zero pack of the bundle
+		if [ "$DEBUG" == true ]; then
+			echo "dangling link -> $bundle_link"
+		fi
+		if [ "$local_bundle" = true ]; then
+			sudo mkdir -p "$target_path$(dirname "$val")"
+			# if local_bundle is enabled since we cannot copy a bad link create a new one
+			# in the appropriate location in target-dir with the corrent name
+			sudo ln -s "$(generate_random_name /does_not_exist-)" "$target_path$val"
+		fi
+	done
+
+	# 6) Add the bundle to the MoM
+	add_to_manifest "$version_path"/Manifest.MoM "$manifest" "$bundle_name"
+
```

Comment:
Adding `sign_manifest "$version_path"/Manifest.MoM` here gets another 25 tests to pass